microdi-js.d.ts

```diff
@@ -18,6 +18,7 @@ export class Container {
     createChildContainer():Container;
     register(name:String, proto:any):RegistrationModifier;
     registerInstance<T>(name:String, instance:T, isExternallyOwned?):void;
+    registerFactory<T>(name:String, factory:(contextContainer:Container) => T):RegistrationModifier;

```

Comment:
I think this should be `factory:(container:Container, ...args:any[]) => T)` 
i.e. rename `contextContainer`, to `context`, add `...args:any[]`